siva/storage.go

```diff
@@ -185,3 +255,190 @@ func getSivaFS(
 
 	return sivafs.NewFilesystem(base, path, sivaTmpFS)
 }
+
+func newRefStorage(fs billy.Filesystem) (memory.ReferenceStorage, error) {
+	rs := map[plumbing.ReferenceName]*plumbing.Reference{}
+	head, err := readReferenceFile(fs, headRefPath, plumbing.HEAD.String())
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+
+		head = plumbing.NewSymbolicReference(plumbing.HEAD, plumbing.Master)
+	}
+
+	rs[head.Name()] = head
+
+	refs, err := findPackedRefs(fs)
```

Comment:
We should also read unpacked refs in case they exist. Maybe using `DotGit.Refs` from go-git.